src/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs

```diff
@@ -154,15 +154,31 @@ public static void JsonIgnoreAttribute_UnsupportedCollection()
                         }
                     }";
 
-            // Unsupported collections will throw by default.
+            // Unsupported collections will throw on deserialize by default.
             Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<ClassWithUnsupportedDictionary>(json));
+            
             // Using new options instance to prevent using previously cached metadata.
             JsonSerializerOptions options = new JsonSerializerOptions();
-            Assert.Throws<NotSupportedException>(() => JsonSerializer.Serialize(new ClassWithUnsupportedDictionary(), options));
+            string serialized = JsonSerializer.Serialize(new ClassWithUnsupportedDictionary(), options);
+            
+            // Object keys are fine on serialization if the keys are strings.
```

Comment:
I see. Perhaps this was the reason for some of my other comments about System.Object usage?

Reply:
Hmm I don't think they're related. This change is because we now only throw NSE when we are about deserialize, but can't (no longer during warm up which prevents serialization)